bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Modal.java

```diff
@@ -0,0 +1,104 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.models;
+
+import org.jetbrains.annotations.NotNull;
+import org.osgi.annotation.versioning.ConsumerType;
+
+import com.adobe.cq.export.json.ComponentExporter;
+
+/**
+ * Defines the {@code Modal} Sling Model used for the
+ * {@code /apps/core/wcm/components/modal} component. This component currently
+ * supports using "#{modalId}" in the url
+ *
+ * @since com.adobe.cq.wcm.core.components.models 12.8.0
+ */
+@ConsumerType
+public interface Modal extends ComponentExporter {
+
+	/**
+	 * Returns the modal id based on the hash generated from the component path
+	 * 
+	 * @return modelId
+	 * @since com.adobe.cq.wcm.core.components.models 12.8.0
+	 */
+	default String getModalId() {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * Returns the description for the modal
+	 * 
+	 * @return description
+	 * @since com.adobe.cq.wcm.core.components.models 12.8.0
+	 */
+	default String getDescription() {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * Returns the boolean value defining whether to show modal by default on page
+	 * load or not
+	 * 
+	 * @return defaultModalShow
+	 * @since com.adobe.cq.wcm.core.components.models 12.8.0
+	 */
+	default boolean getShowModalByDefault() {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * Returns the type of the fragment chosen
+	 * 
+	 * @return fragmentType
+	 * @since com.adobe.cq.wcm.core.components.models 12.8.0
+	 */
+	default String getFragmentType() {

```

Comment:
This is undocumented and not really used, as we could have just a `path` instead of `contentFragmentPath` and `experienceFragmentPath`. I don't see any difference in how we handle the two cases in the edit dialog or the actual component rendering, other than the default browsing path (which can now be just `/content`).

Reply:
Had a discussion with @gitrajat , we will keep the `type` property for now as it makes it easier to switch the initial browsing paths between `/content/dam` and `/content/experience-fragments`. The return type will be an enumeration though so that it's easier to foresee possible values.

Reply:
@vladbailescu  Updating this comment.
As per our discussion, there will be only 1 pathfield. Changing "rootpath" will be taken care by custom JS.